### DISCUSSION TITLE
Custom rendering engine for Jupyter Notebooks

### DISCUSSION THREAD

**Discussion by pplonski** [community/community#55175] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

There about 4 millions of notebooks rendered on the GitHub every day. Notebook itself is a combination of code and outputs saved as a JSON file. Right now, all notebooks are converted to HTML when displayed. It would be useful to have an option to customise the rendering engine. For example, I would like to hide code and just display outputs in rendered view.

**Comment by Jordan99990** [comment]
You could create a css file and make a div with display property none and after you create your notebook just check the raw JSON for url and add at the add of the url '?raw=true' and open it in a new tab.
It will look something like this
%%html
<kink rel="stylesheet" type="text/css" href="[https://raw.github....com/{username}/{repo}/{branch}/file.css>](https://raw.github....com/%7Busername%7D/%7Brepo%7D/%7Bbranch%7D/file.css%3E)
replace the values inside {} with your Github username, name of your repository where the "file.css" is located and the branch name and lastly save the notebook.

**Reply by Jordan99990** [reply]
It would "simulate" at least the hiding feature that you mentioned

**Reply by pplonski** [reply]
Thank you @Jacques9, I would like to have generic solution provided by GitHub service. 

I think the problem is very broad, because there might be many different files that can have custom render engines.

**Comment by albas99** [comment]
I think this is great idea, and even though @Jacques9 suggestion might provide a workaround, it's not necessarily the best way to go, especially for people who use notebooks for their main workflow, I think Github providing a more generic solution is the best way to go. I think even if they don't provide an option for custom rendering, just having it render the way it's supposed to look(as a notebook with cells and outputs displayed correctly(similar to how DagsHub does it)) would go a long way because the way it renders right now is beyond terrible.
